### PR TITLE
Support authentication on artifacts download location

### PR DESCRIPTION
**Pull request by vdotjansen**
Many servers cannot access the internet directly. I would like our ci server to be able to use mvnw. We use a maven repository (artifactory) to cache and validate artifacts we download. our artifactory is password protected and has different rights based on the users. I have added authentication based on environment variables as including the username and password in the URL would exposes our user and password with way too much rights.

### CODE REVIEW

File: mvnw [hunk 1]
```diff
@@ -225,12 +225,21 @@ else
...
+        else
+            wget --http-user=$MVNW_WRAPPER_AUTH_USERNAME --http-password=$MVNW_WRAPPER_AUTH_PASSWORD "$jarUrl" -O "$wrapperJarPath"
+        fi
     elif command -v curl > /dev/null; then
         if [ "$MVNW_VERBOSE" = true ]; then
           echo "Found curl ... using curl"
         fi
-        curl -o "$wrapperJarPath" "$jarUrl"
+        if [ -z "$MVNW_WRAPPER_AUTH_USERNAME" ] || [ -z "$MVNW_WRAPPER_AUTH_PASSWORD" ]; then
+            curl -o "$wrapperJarPath" "$jarUrl" -f
```
Comment (vdotjansen) [inline mvnw:241]: Added -f to not create a file when the http status is not correct (in my example 401).

**Comment by sanderino666** [comment]
I could really use this feature for our company. Here is my upvote!

**Comment by mosabua** [comment]
Could you resolve the conflict so we get this working with TLS 1.2 on Windows (I cant test or develop this on Windows. 

Also could you simplify the property names to just MVNW_USERNAME and MVNW_PASSWORD?

Last but not least... does this basically replace https://github.com/takari/maven-wrapper/pull/96 .. it seems like this is a much more complete implementation. Am I right or am I missing something?

**Comment by mosabua** [comment]
Once the above items are covered I am hoping to cut a new release soon.

**Pull request merged**